.github/workflows/versionincrement.yaml

```diff
@@ -0,0 +1,78 @@
+---
+name: Version Increment
+
+on:
+  workflow_dispatch:
+    inputs:
+      logLevel:
+        description: Log level
+        required: true
+        default: warning
+        type: choice
+        options:
+          - info
+          - warning
+          - debug
+
+jobs:
+  plugin-version-increment-sync:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        repo: [common-utils, job-scheduler, ml-commons, performance-analyzer, performance-analyzer-rca, security, alerting, index-management, sql, observability, asynchronous-search, k-NN, anomaly-detection, cross-cluster-replication, dashboards-reports, geospatial, notifications]
+        branch: [main, 2.x, '2.1', '2.2']
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - name: Check out OpenSearch repo
+        uses: actions/checkout@v3
+        with:
+          repository: opensearch-project/OpenSearch
+          ref: ${{ matrix.branch }}
+      - name: Fetch OpenSearch version
+        run: |
+          OPENSEARCH_VERSION=$(cat buildSrc/version.properties | grep opensearch | cut -d= -f2 | grep -oE '[0-9.]+')
+          OPENSEARCH_VERSION=$OPENSEARCH_VERSION-SNAPSHOT
+          echo "OPENSEARCH_VERSION=$OPENSEARCH_VERSION" >> $GITHUB_ENV
+          echo "OPENSEARCH_VERSION_NUMBER=$OPENSEARCH_VERSION" | cut -f1 -d"-" >> $GITHUB_ENV
+      - name: Check out plugin repo
+        uses: actions/checkout@v3
+        with:
+          repository: opensearch-project/${{ matrix.repo }}
+          ref: ${{ matrix.branch }}
+      - name: Version Increment plugin repo
+      # If condition for plugins with Gradle Project in subfolders.
+        run: |
+          echo "OpenSearch Core repo version on branch ${{ matrix.branch }} is ${{ env.OPENSEARCH_VERSION_NUMBER }}"
+          if [ ${{ matrix.repo }} == "dashboards-reports" ]; then
```

Comment:
Have you tried having a single matrix like so:

```
{ repo: 'common-utils' },
{ repo: 'dashboard-reports', path: 'reports-scheduler' },
...
```

and combine it with the branch matrix to avoid all these `if`'s?

Reply:
Let me try this @dblock, but still we might need one `if` condition to check if `path` exists and cd to the `path`, I will explore this.